### DISCUSSION TITLE
Load CefSharp from an absolute path

### DISCUSSION THREAD

**Discussion by stadlert** [cefsharp/CefSharp#3932] [General]
Hello!

Is it possible to load the CefSharp files at runtime from an absolute path (e.g. C:\cefsharp\) instead of the build path (or its subfolders)?

The reason why I want do this is the following:
I'm developing an WPF UserControl which shows a CefSharp browser window. This UserControl is called from an external proprietary application (with an XAML file) which has some special behaviour: 
The application does not load the files from the original path but copies it to another location first. Unfortunately, only dll files are copied to the new location, all other file types (e.g. CefSharp.BrowserSubprocess.exe) are ignored.
So if I could load CefSharp from a static path (e.g. C:\cefsharp\) where I copy the CefSharp files once manually, it would make the integration into my application much easier.

I hope I have not overlooked an appropriate instruction.

Thank you and best regards,
Thomas

**Comment by amaitland** [comment]
Depends on what version of .Net are you targeting, if you are using 4.x then you can use https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.assemblyresolve?view=net-6.0  event, just make sure to subscribe before any attempt is made to load the assemblies. Module initializer might come in handy here. 

See https://github.com/cefsharp/CefSharp/blob/1fb3f17599a4805772f398d2adb1892b5dfd0fc4/CefSharp/CefRuntime.cs#L33 which should give you a rough idea of what's required. You can create your own method to suite your requirements. 

> The application does not load the files from the original path but copies it to another location first

Is the application using shadow copy? CefSharp will only if loaded into the default application domain.

**Comment by stadlert** [comment]
Hi!

I use .NET 4.5.2 and this solution works very well for me - thank you!

The application is not using shadow copies, but I don't know the reason why at runtime start the referenced assemblies are copied from the original foder to a cache folder...

Best regards,
Thomas
